src/contracts/interfaces.ts

```diff
@@ -32,98 +55,89 @@ export function getContractClassHash(name: string) {
-
-  files.forEach((file) => {
-    if (file.endsWith(".json")) {
-      const filePath = path.join(dirPath, file);
-      const fileContent = fs.readFileSync(filePath, "utf-8");
-      const fileName = path.basename(file, ".json");
-      jsonObjects[fileName] = JSON.parse(fileContent);
-      const classHash = getContractClassHash(fileName);
-      if (!classHash) {
-        throw new Error(`Provide class hash for the contract ${fileName}! It is stored in the artifacts folder but the class hash is missing, therefore it cannot be deployed.`);
-      }
-
-    }
-  });
-
-  return jsonObjects;
-}
-
-const contractArtifacts = importAllJsonContractsArtifacts();
-
-type ContractsInterfacesType = {
-  [key: string]: {
-    abi: Abi;
-    sierra_program_debug_info: object,
-    contract_class_version: string,
-    entry_points_by_type: object,
-  };
-};
-
-export const throwErrorForWrongContractId = (contractName: any) => {
-  if (!contractsInterfaces[contractName]) {
-    throw new Error(`Invalid contractName: ${contractName}. Supported contracts can be checked by calling endpoint /api/public/contracts`);
+const getContractDescription = (contractName: string) => {
+  switch (contractName) {
+    case contractsNames().EntranceContract:
+      return "🚨 FIlL THIS OUt";
+    case contractsNames().ERC1155EventTicket:
+      return "Multi-token standard for creating 'tickets on steroids'. Can be limited to whitelisted marketplaces, collect royalties on the secondary market, used for fan rewards & perks, kill bots  and offer endless  advanced functionality beyond traditional ticketing systems.";
+    case contractsNames().ERC20EventCurrency:
+      return "Token standard for creating in-house, event-specific, or in-app currencies. Ideal for community building and native event payments, enhancing user engagement and providing a seamless transaction experience within our ecosystem.";
+    default:
+      throw new Error(`Provide description for the contract ${contractName}!`);
   }
 };
 
-export const getContractsFunctions = (contractName: any) => {
-  throwErrorForWrongContractId(contractName);
+export const getContractsFunctions = (contractName: any, convertFunctionTypesFromCairo = false) => {
+  throwErrorForWrongContractName(contractName);
   const abi = contractsInterfaces[contractName].abi;
-  const topLevelFunctions = abi.filter(item => item.type === "function");
+  const topLevelFunctions = abi
+    .filter(item => item.type === "function");
   const interfaceFunctions = abi
     .filter(item => item.type === "interface")
     .flatMap(interfaceItem => interfaceItem.items.filter(i => i.type === "function"));
 
-  return [...topLevelFunctions, ...interfaceFunctions]
+  const allFunctions = [...topLevelFunctions, ...interfaceFunctions];
+
+  return allFunctions
     .map((i: any) => ({
       name: i.name,
-      inputs: i.inputs,
+      inputs: convertFunctionTypesFromCairo ? cairoInputsFormat(i.inputs) : i.inputs,
```

Comment:
in case of gasless tx, functions cannot be formatted to more human readable form, ergo I've parametrized the function here